src/shared/components/ResourceEditor/index.tsx

```diff
@@ -173,22 +185,30 @@ const ResourceEditor: React.FunctionComponent<ResourceEditorProps> = props => {
   return (
     <div
       data-testid="resource-editor"
-      className={valid ? 'resource-editor' : 'resource-editor _invalid'}
+      className={isValidJSON ? 'resource-editor' : 'resource-editor _invalid'}
     >
       {showControlPanel && (
         <div className="control-panel">
-          <div>
-            {editable && isEditing && valid && (
-              <div className="feedback _positive">
-                <CheckCircleOutlined /> Valid
-              </div>
-            )}
-            {editable && isEditing && !valid && (
-              <div className="feedback _negative">
-                <ExclamationCircleOutlined /> Invalid JSON-LD
-              </div>
-            )}
-          </div>
+          {editable && isEditing && (
+            <>
+              {isValidJSON && (!linterIssues || linterIssues.length === 0) && (
+                <div className="feedback _positive">
+                  <CheckCircleOutlined /> Valid
+                </div>
+              )}
+              {!isValidJSON && (
+                <div className="feedback _negative">
+                  <ExclamationCircleOutlined /> Invalid JSON-LD
+                </div>
+              )}
+              {isValidJSON && linterIssues && linterIssues.length > 0 && (
```

Comment:
Question: According to the type definition `linterIssues` is always an array (right?). If so, why/when do we need to check if it is truthy (I mean this part of the check `..&& linterIssues &&...`)? An array, even if empty, will always be truthy, so do we need this?

The same goes for line 194 above.

Reply:
That's true. I encountered a linter error when I removed the truthy check. However, I can bypass this issue by initialising the state `const [linterIssues, setLinterIssues] = useState<LinterIssue[]>([]);` with an empty array. Fixed it, thanks!